src/cargo/core/shell.rs

```diff
@@ -79,17 +78,15 @@ impl fmt::Debug for Shell {
 /// A `Write`able object, either with or without color support
 enum ShellOut {
     /// A plain write object without color support
-    Write(Box<dyn Write>),
+    Write(AutoStream<Box<dyn Write>>),
     /// Color-enabled stdio, with information on whether color should be used
     ///
     /// The separate buffered fields are used for buffered writing to the
     /// corresponding stream. The non-buffered fields should be used when you
     /// do not want content to be buffered.
     Stream {
-        stdout: StandardStream,
-        buffered_stdout: BufferWriter,
-        stderr: StandardStream,
-        buffered_stderr: BufferWriter,
+        stdout: AutoStream<std::io::Stdout>,
+        stderr: AutoStream<std::io::Stderr>,
         stderr_tty: bool,
         color_choice: ColorChoice,
     },
```

Comment:
Seems like you removed the buffered fields here. The doc comment might need adjusting, since it's still referencing "buffered fields."